pkg/ccl/backupccl/restore_job.go

```diff
@@ -281,8 +284,66 @@ func restore(
 	// which are grouped by keyrange.
 	highWaterMark := job.Progress().Details.(*jobspb.Progress_Restore).Restore.HighWater
 
-	importSpans := makeSimpleImportSpans(dataToRestore.getSpans(), backupManifests,
-		backupLocalityMap, introducedSpanFrontier, highWaterMark, targetRestoreSpanSize.Get(execCtx.ExecCfg().SV()))
+	var importSpans []execinfrapb.RestoreSpanEntry
+	var checkpointingSpanFrontier *spanUtils.Frontier
+	if execCtx.ExecCfg().Settings.Version.IsActive(restoreCtx, clusterversion.V23_1Start) {
```

Comment:
we ought to check the `r.job.Payload().CreationClusterVersion` instead of the version bound to the execCfg() to avoid the following situation:

- user begin restore in 22.2
- user pauses restore
- user upgrades
- user resumes. progress doesn't make sense.